src/ui-layout.js

```diff
@@ -796,6 +807,37 @@ angular.module('ui.layout', [])
             if(!element.hasClass('stretch')) element.addClass('stretch');
             if(!element.hasClass('ui-layout-container')) element.addClass('ui-layout-container');
 
+            scope.$watch('collapsed', function (val, old) {
+              if (old !== undefined && val !== old) {
+                var index = scope.container.index;
+                var splitter = ctrl.containers[index + 1],
+                  el,
+                  beforeContainer,
+                  afterContainer;
+
+                if (splitter) {
+                  el = splitter.element[0].children[0];
+                  beforeContainer = scope.container;
+                  afterContainer = ctrl.containers[index + 2];
+
+                } else {
+                  splitter = ctrl.containers[index - 1];
+                  el = splitter.element[0].children[1];
+                  beforeContainer = ctrl.containers[index - 2];
+                  afterContainer = scope.container;
+                }
+
+
+                $timeout(function(){
```

Comment:
Aha. This is the trick I know how to get around "$digest in progess" error, which was thrown after trying to trigger the click. Not sure how to use $q here.